### PR TITLE
Clarify verification on multiple signatures in a envelope

### LINKED ISSUE #40: How to verify an envelope properly?
The [protocol](https://github.com/secure-systems-lab/dsse/blob/master/protocol.md) specifies how to verify a single signature but does not specify how to verify signatures in an envelope.

The reference implementation in different repos varies:
- [secure-systems-lab/dsse](https://github.com/secure-systems-lab/dsse/tree/master/implementation) in python considers that an envelope is valid if one of the signatures is valid.
- [in-toto/in-toto-golang](https://github.com/in-toto/in-toto-golang/tree/master/pkg/ssl) in golang considers that an envelope is valid if all of the signatures are valid.

Should the DSSE spec define the verification process of an envelope or explicitly call out that the verification process is purely defined by the implementer?

Additionally, for the golang implementation, the adversary can make the verification pass if some signatures in the envelope are valid simply by removing the invalid sigantures out.

### PR DESCRIPTION
**Pull request by shizhMSFT**
Resolves #40

### CODE REVIEW

**Comment by trishankatdatadog** [comment]
Maybe add a bit more text please. You could mention TUF or in-toto as applications which use threshold signatures, where m out of n signatures must be valid for the application to accept the payload.

File: envelope.md [hunk 1]
```diff
@@ -52,6 +52,10 @@ envelopes with individual signatures.
 }
 ```
 
+The verification policy on multiple signatures is application-specific.
+For instances, [TUF] and [in-toto] have a threshold policy where `m` out of `n`
```
Comment (trishankatdatadog) [inline envelope.md:56]:
```suggestion
For instance, [TUF] and [in-toto] have a threshold policy where `m` out of `n`
```

**Review by trishankatdatadog — APPROVED** [review]

**Comment by trishankatdatadog** [comment]
@MarkLodato @SantiagoTorres LGTY?

**Review by MarkLodato — APPROVED** [review]
I think this is an improvement, but IMO it is is not sufficient to resolve #40. In addition, I think we need an update to protocol.md to explain how the verifier should handle multiple signatures. I'll continue discussion in #40.

**Comment by adityasaky** [comment]
Closing this since #45 has been accepted.

**Pull request closed without merging**